.github/workflows/go_build.yaml

```diff
@@ -68,4 +68,93 @@ jobs:
 
     - name: Spec Test Coverage
       run: |
-            go run tests/spectests/cmd/analyzer/analyzer.go
\ No newline at end of file
+            # Output the full report to console.
+            go run tests/spectests/cmd/analyzer/analyzer.go
+
+
+  go_spec_test_badge_update:
+    # change later to on merge: github.event.pull_request.merged
+    # Merged onto the default branch.
+    # later to enable for default branch, add: && github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
+    if: github.event_name == 'pull_request'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+
+    - name: Set up Go 1.22
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.22
+      id: go
+
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: Get dependencies
+      run: |
+            go mod download
+    - name: Build
+      run: |
+            go build ./...
+
+    - name: Setup Env Variables
+      id: vars
+      # env:
+      #   BADGE_DIR: coverage-reports
+      #   BADGE_PATH: coverage-reports/spec-coverage.json
+      #   BRANCH: gh-pages
+      run: |
+            # Save total test coverage to a variable.
+            SPEC_COVERAGE=$( \
+              go run tests/spectests/cmd/analyzer/analyzer.go \
+              | grep 'Not Skipped' --max-count=1 \
+              | cut -d' ' -f 4 \
+              | tr -d "()"
+            )
+            echo "spec_coverage=$SPEC_COVERAGE" >> "${GITHUB_OUTPUT}"
+
+            # BADGE_DIR="coverage-reports"
+            echo "dir=coverage-reports" >> "${GITHUB_OUTPUT}"
+            # BADGE_PATH="coverage-reports/spec-coverage.json"
+            echo "path=coverage-reports/spec-coverage.json" >> "${GITHUB_OUTPUT}"
+            # BRANCH="gh-pages"
+            echo "branch=gh-pages" >> "${GITHUB_OUTPUT}"
+
```

Comment:
as discussed, we can probably simplify this by either:

1. Having the analyzer take a flag to produce the json we need
2. Or instead of generating the json, serving it, and using sheilds.io we can just generate the badge svg directly from the analyzer and statically serve it from github pages (or artifacts)

Here's a library that'll generate the svg directly in go: https://github.com/essentialkaos/go-badge